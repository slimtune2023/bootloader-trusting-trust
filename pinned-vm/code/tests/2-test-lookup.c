/* Miles' little test for tlb_contains_va */

#include "rpi.h"
#include "procmap.h"

void notmain(void) { 
    kmalloc_init_set_start((void*)SEG_HEAP, MB(1));

    // turn on the pinned MMU: identity map.
    procmap_t p = procmap_default_mk(dom_kern);
    procmap_pin_on(&p);

    // if we got here MMU must be working b/c we have accessed GPIO/UART
    trace("hello: mmu must be on\n");

    lockdown_print_entries("before lookup");
    uint32_t result = 0;
    for(int i = 0; i < p.n; i++) {
        uint32_t addr = p.map[i].addr + 0xFAA0;
        if(!tlb_contains_va(&result, addr))
            panic("did not contain valid VA for %x\n", addr);

        if(!tlb_contains_va(&result, addr))
            panic("did not contain valid VA %x\n", addr);
        if(result != addr)
            panic("VA address didn't match!  got=%x, expected=%x", 
                result,addr);
        trace("%x mapped to %x correctly\n", addr, result);
    }
    if(tlb_contains_va(&result, 0xDEADFAA0))
        panic("contained invalid VA\n");

    trace("SUCCESS!\n");
}
