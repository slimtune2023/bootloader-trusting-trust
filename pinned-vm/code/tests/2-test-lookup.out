TRACE:notmain:hello: mmu must be on
TRACE:lockdown_print_entries:-----  <before lookup> ----- 
TRACE:lockdown_print_entries:  pinned TLB lockdown entries:
TRACE:lockdown_print_entry:   idx=0
TRACE:lockdown_print_entry:     va_ent=0x20000200: va=0x20000|G=1|ASID=0
TRACE:lockdown_print_entry:     pa_ent=0x200000c3: pa=0x20000|nsa=0|nstid=0|size=11|apx=1|v=1
TRACE:lockdown_print_entry:     attr=0x80: dom=1|xn=0|tex=0|C=0|B=0
TRACE:lockdown_print_entry:   idx=1
TRACE:lockdown_print_entry:     va_ent=0x20100200: va=0x20100|G=1|ASID=0
TRACE:lockdown_print_entry:     pa_ent=0x201000c3: pa=0x20100|nsa=0|nstid=0|size=11|apx=1|v=1
TRACE:lockdown_print_entry:     attr=0x80: dom=1|xn=0|tex=0|C=0|B=0
TRACE:lockdown_print_entry:   idx=2
TRACE:lockdown_print_entry:     va_ent=0x20200200: va=0x20200|G=1|ASID=0
TRACE:lockdown_print_entry:     pa_ent=0x202000c3: pa=0x20200|nsa=0|nstid=0|size=11|apx=1|v=1
TRACE:lockdown_print_entry:     attr=0x80: dom=1|xn=0|tex=0|C=0|B=0
TRACE:lockdown_print_entry:   idx=3
TRACE:lockdown_print_entry:     va_ent=0x200: va=0x0|G=1|ASID=0
TRACE:lockdown_print_entry:     pa_ent=0xc3: pa=0x0|nsa=0|nstid=0|size=11|apx=1|v=1
TRACE:lockdown_print_entry:     attr=0x88: dom=1|xn=0|tex=1|C=0|B=0
TRACE:lockdown_print_entry:   idx=4
TRACE:lockdown_print_entry:     va_ent=0x100200: va=0x100|G=1|ASID=0
TRACE:lockdown_print_entry:     pa_ent=0x1000c3: pa=0x100|nsa=0|nstid=0|size=11|apx=1|v=1
TRACE:lockdown_print_entry:     attr=0x88: dom=1|xn=0|tex=1|C=0|B=0
TRACE:lockdown_print_entry:   idx=5
TRACE:lockdown_print_entry:     va_ent=0x8f00200: va=0x8f00|G=1|ASID=0
TRACE:lockdown_print_entry:     pa_ent=0x8f000c3: pa=0x8f00|nsa=0|nstid=0|size=11|apx=1|v=1
TRACE:lockdown_print_entry:     attr=0x88: dom=1|xn=0|tex=1|C=0|B=0
TRACE:lockdown_print_entry:   idx=6
TRACE:lockdown_print_entry:     va_ent=0x7f00200: va=0x7f00|G=1|ASID=0
TRACE:lockdown_print_entry:     pa_ent=0x7f000c3: pa=0x7f00|nsa=0|nstid=0|size=11|apx=1|v=1
TRACE:lockdown_print_entry:     attr=0x88: dom=1|xn=0|tex=1|C=0|B=0
TRACE:lockdown_print_entry:   idx=7
TRACE:lockdown_print_entry:     
TRACE:lockdown_print_entries:----- ---------------------------------- 
TRACE:notmain:0x2000faa0 mapped to 0x2000faa0 correctly
TRACE:notmain:0x2010faa0 mapped to 0x2010faa0 correctly
TRACE:notmain:0x2020faa0 mapped to 0x2020faa0 correctly
TRACE:notmain:0xfaa0 mapped to 0xfaa0 correctly
TRACE:notmain:0x10faa0 mapped to 0x10faa0 correctly
TRACE:notmain:0x8f0faa0 mapped to 0x8f0faa0 correctly
TRACE:notmain:0x7f0faa0 mapped to 0x7f0faa0 correctly
TRACE:notmain:SUCCESS!
